src/main/java/dev/murad/shipping/entity/accessor/EnergyHeadVehicleDataAccessor.java

```diff
@@ -17,69 +19,31 @@ public EnergyHeadVehicleDataAccessor(ContainerData data) {
      */
 
     public int getEnergy() {
-        int lo = this.data.get(1) & SHORT_MASK;
-        int hi = this.data.get(2) & SHORT_MASK;
+        int lo = this.data.get(15) & SHORT_MASK;
+        int hi = this.data.get(16) & SHORT_MASK;
         return lo | hi << 16;
     }
 
     public int getCapacity() {
-        int lo = this.data.get(3) & SHORT_MASK;
-        int hi = this.data.get(4) & SHORT_MASK;
+        int lo = this.data.get(17) & SHORT_MASK;
+        int hi = this.data.get(18) & SHORT_MASK;
         return lo | hi << 16;
     }
 
-    public boolean isLit() {
-        return this.data.get(5) == 1;
-    }
-
-    public boolean isOn() {
-        return this.data.get(6) == 1;
-    }
-
-    public int visitedSize() {
-        return this.data.get(7);
-    }
-
-    public int routeSize() {
-        return this.data.get(8);
-    }
-
-    public static class Builder {
-        SupplierIntArray arr;
-
-        public Builder(int uuid) {
-            this.arr = new SupplierIntArray(9);
-            this.arr.set(0, uuid);
+    public static class Builder extends HeadVehicleDataAccessor.Builder {
+        public Builder() {
+            this.arr = new SupplierIntArray(20);
         }
 
         public Builder withEnergy(IntSupplier energy) {
-            this.arr.setSupplier(1, () -> energy.getAsInt() & SHORT_MASK);
-            this.arr.setSupplier(2, () -> (energy.getAsInt() >> 16) & SHORT_MASK);
+            this.arr.setSupplier(15, () -> energy.getAsInt() & SHORT_MASK);
```

Comment:
kinda confused, why is this now higher?

Reply:
Decided to have all the common attributes go first, and move energy/steam specific ones further out for future expansion. Got tired of refactoring everything every time we add something